From 09268fa4556d453348c0f6cf7f931afe9bf91a31 Mon Sep 17 00:00:00 2001
From: Flora Gao <b13318@freescale.com>
Date: Tue, 13 Jan 2009 16:11:20 +0800
Subject: [PATCH] ENGR00104171 MX35: FlexCan GPIO pad setting updates

1.Modify the pad setting for CAN:
    remove PAD_CTL_ODE_OpenDrain,
    add PAD_CTL_DRV_HIGH.
2.Modify the setting of flexcan_data.core_reg to "SW1".

Signed-off-by: Flora Gao <b13318@freescale.com>
---
 arch/arm/mach-mx35/mx35_3stack.c      |    4 ++--
 arch/arm/mach-mx35/mx35_3stack_gpio.c |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-mx35/mx35_3stack.c b/arch/arm/mach-mx35/mx35_3stack.c
index 2b848a7..012d12e 100644
--- a/arch/arm/mach-mx35/mx35_3stack.c
+++ b/arch/arm/mach-mx35/mx35_3stack.c
@@ -842,8 +842,8 @@ static void mx35_3stack_fixup_for_board_v1(void)
 	mxc_bt_data.bt_vdd = "SW1";
 
 #if defined(CONFIG_CAN_FLEXCAN) || defined(CONFIG_CAN_FLEXCAN_MODULE)
-	flexcan_data[0].core_reg = "GPO2";
-	flexcan_data[1].core_reg = "GPO2";
+	flexcan_data[0].core_reg = "SW1";
+	flexcan_data[1].core_reg = "SW1";
 #endif
 }
 
diff --git a/arch/arm/mach-mx35/mx35_3stack_gpio.c b/arch/arm/mach-mx35/mx35_3stack_gpio.c
index f58b521..0c1cc46 100644
--- a/arch/arm/mach-mx35/mx35_3stack_gpio.c
+++ b/arch/arm/mach-mx35/mx35_3stack_gpio.c
@@ -1295,7 +1295,7 @@ void gpio_can_active(int id)
 	switch (id) {
 	case 0:
 		pad = PAD_CTL_HYS_SCHMITZ | PAD_CTL_PKE_ENABLE | \
-		    PAD_CTL_PUE_PUD | PAD_CTL_100K_PU | PAD_CTL_ODE_OpenDrain;
+		    PAD_CTL_PUE_PUD | PAD_CTL_100K_PU | PAD_CTL_DRV_HIGH;
 		mxc_request_iomux(MX35_PIN_I2C2_CLK, MUX_CONFIG_ALT1);
 		mxc_request_iomux(MX35_PIN_I2C2_DAT, MUX_CONFIG_ALT1);
 		mxc_iomux_set_pad(MX35_PIN_I2C2_CLK, pad);
-- 
1.5.4.4

